### PR TITLE
i hope it works (catalog)

### PR DESCRIPTION
**Pull request by plankton9991**
- [DEMO LINK](https://plankton9991.github.io/layout_catalog/)
- [TEST REPORT LINK](https://plankton9991.github.io/layout_catalog/report/html_report/)
Відступ вертикальний має бути однаковий, але в тесті і макеті він різний

### CODE REVIEW

File: src/styles/blocks/main/card.scss [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    margin-bottom: 17px;
+  }
+
+  &__price {
+    font-size: 12px;
+    line-height: 18px;
+    color: var(--grey-color);
+  }
+
+  &__price-number {
```
Comment (Vakolyyk) [inline src/styles/blocks/main/card.scss:46]: Вложи в &__price

**Review by Vakolyyk — CHANGES_REQUESTED** [review]
Тести не підгрузились, подивись п.11 в чеклісті. В тебе при видаленні однієї карти вони встають по центрі

**Comment by plankton9991** [comment]
Відступ вертикальний має бути однаковий, але в тесті і макеті він різний

File: src/index.html [hunk 2]
```diff
@@ -4,9 +4,360 @@
...
+              href="#apple"
+            >Apple</a>
+            <a class="header__nav-link" href="#samsung">Samsung</a>
+            <a class="header__nav-link" href="#phones">Смартфоны и телефоны</a>
+            <a
+              class="header__nav-link"
+              href="#laptops"
+              data-qa="nav-hover"
+            >Ноутбуки и компьютеры
+            </a>
```
Comment (illiarerun) [inline src/index.html:35]:
```suggestion
            >
              Ноутбуки и компьютеры
            </a>
```

File: src/styles/normalize.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+p,
+h2 {
+  margin: 0;
+}
+
+html {
+  font-family: Roboto, sans-serif;
+
+  --blue-color: #060b35;
+  --grey-color: #616070;
```
Comment (illiarerun) [inline src/styles/normalize.scss:10]: you can use scss variables

File: src/styles/catalog.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+}
+
+.products__card-container {
+  padding: 50px 0 0;
+  margin: 0 auto;
+  min-width: 944px;
```
Comment (illiarerun) [inline src/styles/catalog.scss:19]: don't hardcode width like that, the layout should be adaptive, use max-width instead, calculate it with using card-width, gap and number of cards

File: src/styles/blocks/main/card.scss [hunk 5]
```diff
@@ -0,0 +1,96 @@
+.card {
+  box-sizing: border-box;
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:2]: box-sizing should be same for all elements, use * selector for that

File: src/styles/blocks/main/card.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+
+  padding: 32px 16px 16px;
+  margin-left: 48px;
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:7]: don't add margins to blocks, create mix (element inside another block for that)

File: src/styles/blocks/main/card.scss [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  max-width: 200px;
+  display: flex;
+  flex-direction: column;
+
+  &:first-child {
+    margin-left: 0;
+  }
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:14]: with this, too

File: src/styles/blocks/main/card.scss [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+  &__title {
+    font-size: 12px;
+    line-height: 18px;
+    font-weight: 500;
+    max-width: 166px;
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:20]: redundant, you have paddings in card for that
```suggestion
```

File: src/styles/blocks/main/card.scss [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+      background-color: #fff;
+      color: #00acdc;
+      border-color: rgba(0,172,220,1);
+    }
+  }
+
+  &__stars {
+    display: flex;
+    flex-direction: row;
+  }
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:82]: that should be style for stars block, not element, and they should be in stars.scss

**Review by illiarerun — CHANGES_REQUESTED** [review]
Nice start, let's work on fixes now:

File: src/index.html [hunk 10]
```diff
@@ -4,9 +4,360 @@
...
+            alt="moyo_logo"
+            class="header__image"
+          >
+        </a>
+        <nav class="header__nav">
+          <ul class="header__list">
+            <a
+              class="header__nav-link header__nav-link--active is-active"
+              href="#apple"
+            >Apple</a>
```
[the remark this answers is not in the document]
Comment (illiarerun) [reply src/index.html:27]: please pay attention to all comments, fix formatting here

File: src/index.html (hunk 2 above)
Comment (illiarerun) [reply src/index.html:35]: please pay attention to all comments, fix formatting here

File: src/styles/products.scss [hunk 11]
```diff
@@ -0,0 +1,23 @@
+$width: calc((200px * 4) + (48px * 3));
```
Comment (illiarerun) [inline src/styles/products.scss:1]: what if width of card changes, or their number, or a gap? store that values in variables too

File: src/styles/products.scss [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  &__card {
+    margin-left: 48px;
+
+    &:first-child {
+      margin-left: 0;
+    }
+
+    &-container {
+      padding: 50px 0 0;
+      max-width: #{width};
```
Comment (illiarerun) [inline src/styles/products.scss:17]: it's more convenient to use it like that, also easier to look for it in code
```suggestion
      max-width: $width;
```

File: src/styles/blocks/main/card.scss [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+  &__link {
+    box-sizing: border-box;
+    text-decoration: none;
+    text-transform: uppercase;
+    color: #fff;
+    font-size: 14px;
+    font-weight: 700;
+    padding: 10px 57px 11px 55px;
```
Comment (illiarerun) [inline src/styles/blocks/main/card.scss:63]: you can just center the text inside, don't use these weird padding values

File: src/styles/blocks/header.scss [hunk 14]
```diff
@@ -0,0 +1,63 @@
+* {
+  box-sizing: border-box;
+}
```
Comment (illiarerun) [inline src/styles/blocks/header.scss:3]: that's a normalize style, put it in normalize.scss

File: src/styles/blocks/header.scss [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+body {
+  margin: 0;
+}
```
Comment (illiarerun) [inline src/styles/blocks/header.scss:7]: that a reset style

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 16]
```diff
@@ -4,9 +4,363 @@
...
+            <div class="card__price-container">
+              <div class="card__price">Цена:</div>
+              <p class="card__price-number">69 999 грн</p>
+            </div>
+
+            <a
+              class="card__link"
+              href="#"
+              data-qa="card-hover"
+            >Купить</a>
```
Comment (AlinaMatuschak) [inline src/index.html:85]:
```suggestion
            >
              Купить
            </a>
```

File: src/styles/blocks/main/card.scss [hunk 17]
```diff
@@ -0,0 +1,85 @@
+$blue-color: #060b35;
+$grey-color: #616070;
```
Comment (AlinaMatuschak) [inline src/styles/blocks/main/card.scss:2]: create separate file for variables

File: src/styles/blocks/main/stars.scss [hunk 18]
```diff
@@ -0,0 +1,29 @@
+.stars__star {
+  height: 16px;
+  width: 16px;
+  margin-right: 4px;
+  background-image: url(../../../images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center center;
+}
+
+.stars--4 .stars__star:nth-child(-n+4) {
```
Comment (AlinaMatuschak) [inline src/styles/blocks/main/stars.scss:10]: you should add all selectors, from 0 to 5, use loop for that

File: src/styles/blocks/main/stars.scss [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+  height: 16px;
+  width: 16px;
+  margin-right: 4px;
```
Comment (AlinaMatuschak) [inline src/styles/blocks/main/stars.scss:13]: you already add this on stars__star class

File: src/styles/blocks/main/stars.scss [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+  background-image: url(../../../images/star-active.svg);
+}
+
+.card {
```
Comment (AlinaMatuschak) [inline src/styles/blocks/main/stars.scss:17]: create separate file for card

File: src/styles/products.scss [hunk 21]
```diff
@@ -0,0 +1,28 @@
+$card-width: 200px;
+$card-numbers: 4;
+$gap: 48px;
+$gap-numbers: $card-numbers - 1;
+
+$width: calc(($card-width * $card-numbers) + ($gap * 3));
```
Comment (AlinaMatuschak) [inline src/styles/products.scss:6]: Separate file

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]
Almost perfect 👍

File: src/index.html [hunk 22]
```diff
@@ -4,9 +4,377 @@
...
+              class="card__image"
+            >
+
+            <h2 class="card__title">
+              Моноблок APPLE A1419 iMac 27" Retina 5K (MNED2UA/A)
+            </h2>
+
+            <div class="card__subtitle">
+              Код товара: 195434
+            </div>
```
Comment (AlinaMatuschak) [inline src/index.html:62]: use semantic tag for text

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**